### DISCUSSION TITLE
Global JS in a module

### DISCUSSION THREAD

**Discussion by andreatosato** [OrchardCMS/OrchardCore#6959] [Q&A] [answered]
I need to add event listener on "[postmessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage)" HTML5 in a module.
Can I add some JS end plugin on index page? Can execute js on each page?

I don't want to implement this on theme because this JS must related with only my module.
Module use thirdy party software, so I can't rewrite postmessage.

Thanks a lot

**Comment by Piedone** [comment, marked as answer]
I converted your issue into a discussion.

You can do this e.g. with a [filter](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Filters/ResourceInjectionFilter.cs) or check out [this library](https://github.com/Lombiq/Helpful-Libraries/blob/dev/Docs/ResourceManagementLibraries.md).

**Reply by andreatosato** [reply]
Thanks @Piedone it works very well!

**Reply by Piedone** [reply]
Great then! You can mark this as an answer so others can see this topic is completed.
